core_contracts/governance/governance.py

```diff
@@ -138,15 +144,19 @@ def castVote(self, name: str, vote: bool) -> None:
         total_for_votes = proposal.total_for_votes.get()
         total_against_votes = proposal.total_against_votes.get()
         if vote:
+            total_for_voters_count = proposal.for_voters_count.get()
             proposal.for_votes_of_user[sender] = stake
             proposal.against_votes_of_user[sender] = 0
             total_for = total_for_votes + stake - prior_vote[0]
             total_against = total_against_votes - prior_vote[1]
+            proposal.for_voters_count.set(total_for_voters_count + 1)
```

Comment:
We need to check if the person has previously voted. If they are making the same vote we just return. If they are changing their vote we remove one from one total and add one to the other.